src/Microsoft.DotNet.Darc/DarcLib/Helpers/AzureAuthentication.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Azure.Core;
+using Azure.Identity;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.Helpers;
+public static class AzureAuthentication
+{
+    public static TokenCredential GetCliCredential() => new ChainedTokenCredential(
+            new AzureCliCredential(),
+            new DefaultAzureCredential());
+
+    public static TokenCredential GetServiceCredential(bool isDevelopment, string? managedIdentityClientId)
+    {
+        if (isDevelopment)
+        {
+            return new DefaultAzureCredential();
+        }
+        else
+        {
+            return new ManagedIdentityCredential(ManagedIdentityId.FromUserAssignedClientId(managedIdentityClientId ??
```

Comment:
Aren't we sometimes using `default` or `system` or something to use the system assigned one? I remember something like that

Reply:
I think we made it a rule that we'll always use the UserAssignedOne, since it gives us more control

Reply:
Okay